scripts/test-dssspell-forge.sh

```diff
@@ -22,12 +23,16 @@ export FOUNDRY_OPTIMIZER=false
 export FOUNDRY_OPTIMIZER_RUNS=200
 export FOUNDRY_ROOT_CHAINID=5
 
-if [[ -z "$MATCH" && -z "$BLOCK" ]]; then
-    forge test --fork-url "$ETH_RPC_URL"
-elif [[ -z "$BLOCK" ]]; then
-    forge test --fork-url "$ETH_RPC_URL" --match-test "$MATCH" -vvv
-elif [[ -z "$MATCH" ]]; then
-    forge test --fork-url "$ETH_RPC_URL" --fork-block-number "$BLOCK"
-else
-    forge test --fork-url "$ETH_RPC_URL" --match-test "$MATCH" --fork-block-number "$BLOCK" -vvv
+EXTRA_ARGS=''
+
+if [[ -n "$MATCH" ]]; then
+    EXTRA_ARGS="${EXTRA_ARGS} -vvv --match-test ${MATCH}"
+elif [[ -n "$NO_MATCH" ]]; then
+    EXTRA_ARGS="${EXTRA_ARGS} -vvv --no-match-test ${NO_MATCH}"
+fi
+
+if [[ -n "$BLOCK" ]]; then
+    EXTRA_ARGS="${EXTRA_ARGS} --fork-block-number ${BLOCK}"
 fi
+
+forge test --fork-url "$ETH_RPC_URL" $EXTRA_ARGS
```

Comment:
Argument passthrough via an `EXTRA_ARGS` env var makes sense in terms of usability.

Security considerations here are that `make test` can now produce unexpected results since `EXTRA_ARGS` could be anything and isn't strictly limited to 'MATCH' and 'BLOCK' like before.

Consider the case where a malicious spell team member wants to make their tests pass / fail / produce certain results. A custom 'EXTRA_ARGS' setup may allow for this while still producing test results which appear genuine.

Of course, someone could always run the tests directly via 'forge test' and do this.

I propose one of two fixes here:

1) EXTRA_ARGS is echoed / provided as output which must be provided alongside passing or failing test results if only to provide context for future spell team members looking back at test results when debugging issues in the future.

2) 'EXTRA_ARGS' is limited to certain env vars which can be passed in (MATCH, NO_MATCH and BLOCK for example) which is the existing method of design.

Overall this is a net improvement in terms of utility. The only thing here is that the 'tests PASS' checklist requirement becomes open ended when the entire set of execution setting arguments are permitted. Limiting the permissible env vars which can be set (old design) is one way of solving this, another way is by changing the checklist such that it mandates `forge test --fork-url "$ETH_RPC_URL"` as the test command being used.

Open to feedback here

Reply:
> 'EXTRA_ARGS' is limited to certain env vars which can be passed in (MATCH, NO_MATCH and BLOCK for example) which is the existing method of design.

It's already like that... Let me add some contextual comments to the code here:

```bash
EXTRA_ARGS='' # whatever value in the EXTRA_ARGS env var is discarded

# if there's a match, add it to extra args
if [[ -n "$MATCH" ]]; then
    # extra args will contain --match-test
    EXTRA_ARGS="${EXTRA_ARGS} -vvv --match-test ${MATCH}"
elif [[ -n "$NO_MATCH" ]]; then
    # extra args will contain --no-match-test
    EXTRA_ARGS="${EXTRA_ARGS} -vvv --no-match-test ${NO_MATCH}"
fi

# if there's a block, add it to extra args
if [[ -n "$BLOCK" ]]; then
    EXTRA_ARGS="${EXTRA_ARGS} --fork-block-number ${BLOCK}"
fi
```

Notice how no other option can be added, unless someone tries to 'inject' stuff like `match='"" --other-option FOO'`, but this can happen anyway, and it's hard to prevent in pure bash script.

Reply:
I agree with these points. Confirmed that the code as present here does in fact limit the env being used by the script, so there is no additional security risk IMO. This was my mistake when interpreting the way it was handling `EXTRA_ARGS`.

That being said, the naming is a bit confusing, can we please have the `EXTRA_ARGS` from env var and the `EXTRA_ARGS` used by the script using different names?

For example, `EXECUTION_ARGS` or `TEST_ARGS` or `INPUT_TEST_ARGS` would be more clear, as this is part of why I was confused initially.